From: Prarit Bhargava <prarit@redhat.com>
Date: Wed, 25 Mar 2009 09:00:44 -0400
Subject: [x86] powernow-k8: export module parameters via sysfs
Message-id: 20090325130043.8122.4874.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL5 PATCH]: powernow-k8: export module parameters via sysfs
Bugzilla: 492010
RH-Acked-by: Brian Maly <bmaly@redhat.com>

Export tscsync and preregister_acpi_perf values via
/sys/modules/powernow-k8/parameters.

Successfully compiled & tested on an AMD system using the powernow-k8 driver.

Resolves BZ 492010.

diff --git a/arch/i386/kernel/cpu/cpufreq/powernow-k8.c b/arch/i386/kernel/cpu/cpufreq/powernow-k8.c
index e7f9744..ce4f5e6 100644
--- a/arch/i386/kernel/cpu/cpufreq/powernow-k8.c
+++ b/arch/i386/kernel/cpu/cpufreq/powernow-k8.c
@@ -1500,8 +1500,8 @@ MODULE_LICENSE("GPL");
 late_initcall(powernowk8_init);
 module_exit(powernowk8_exit);
 
-module_param(tscsync, int, 0);
+module_param(tscsync, int, 0444);
 MODULE_PARM_DESC(tscsync, "enable tsc by synchronizing powernow-k8 changes");
-module_param(preregister_acpi_perf, int, 0);
+module_param(preregister_acpi_perf, int, 0444);
 MODULE_PARM_DESC(preregister_acpi_perf, "allow preregistering of performance"
 		 " related ACPI data");
